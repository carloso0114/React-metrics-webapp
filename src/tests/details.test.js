import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Details from '../components/details.jsx/details';
import store from '../redux/configureStore';

it('renders correctly', () => {
  const fakeData = ['Antigua and Barbuda',
    {
      date: '2021-09-16',
      id: 'antigua_and_barbuda',
      links: [
        {
          href: '/api/2021-09-16/country/antigua_and_barbuda',
          rel: 'self',
          type: 'GET',
        },
      ],
      name: 'Antigua and Barbuda',
      name_es: 'Antigua y Barbuda',
      name_it: 'Antigua e Barbuda',
      regions: [],
      source: 'John Hopkins University',
      today_confirmed: 2304,
      today_deaths: 48,
      today_new_confirmed: 0,
      today_new_deaths: 0,
      today_new_open_cases: 0,
      today_new_recovered: 0,
      today_open_cases: 1017,
      today_recovered: 1239,
      today_vs_yesterday_confirmed: 0,
      today_vs_yesterday_deaths: 0,
      today_vs_yesterday_open_cases: 0,
      today_vs_yesterday_recovered: 0,
      yesterday_confirmed: 2304,
      yesterday_deaths: 48,
      yesterday_open_cases: 1017,
      yesterday_recovered: 1239,
    },
  ];
  const tree = renderer
    .create(<Router><Provider store={store}><Details data={fakeData} /></Provider></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
