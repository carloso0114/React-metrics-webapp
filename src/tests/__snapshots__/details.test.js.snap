// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="sc-gsTEea eVUrAJ navbar"
  >
    <a
      className="sc-dlfnuX habHiA"
      href="/"
      onClick={[Function]}
    >
       &lt; Home
    </a>
    <span
      className="sc-bdfBQB gqyrhZ"
    >
      COVID Analitycs Data
    </span>
    <span>
      <i
        className="bi bi-mic-fill h4 me-3"
      />
      <i
        className="bi bi-gear-fill h4"
      />
    </span>
  </nav>
  <div
    className="container-fluid text-center"
  >
    <h1
      className="sc-eCstlR bOazAA"
    >
      Antigua and Barbuda
    </h1>
    <p
      className="sc-hKgJUU gFGBjw"
    >
      Total deaths:
       
      48
    </p>
    <p
      className="sc-hKgJUU gFGBjw"
    >
      Today new deaths:
       
      0
    </p>
    <p
      className="sc-hKgJUU gFGBjw"
    >
      Today new confirmed cases:
       
      0
    </p>
    <p
      className="sc-hKgJUU gFGBjw"
    >
      Total recovered:
       
      1239
    </p>
    <p
      className="sc-hKgJUU gFGBjw"
    >
      Today new recovered:
       
      0
    </p>
    <p
      className="sc-hKgJUU gFGBjw"
    >
      Source of information:
       
      John Hopkins University
    </p>
  </div>
</div>
`;
